lib/integrations/awsoidc/access_graph_aws_sync.go

```diff
@@ -0,0 +1,113 @@
+)
+
+const (
+	// defaultPolicyNameForTAGSync is the default name for the Inline TAG Policy added to the IntegrationRole.
+	defaultPolicyNameForTAGSync = "AccessGraphAWSSyncAccess"
+)
+
+// AccessGraphAWSIAMConfigureRequest is a request to configure the required Policies to use the TAG AWS Sync.
+type AccessGraphAWSIAMConfigureRequest struct {
+	// IntegrationRole is the Integration's AWS Role used to set up Teleport as an OIDC IdP.
+	IntegrationRole string
+
+	// IntegrationRoleTAGPolicy is the Policy Name that is created to allow access to call AWS APIs.
+	// Defaults to "AccessGraphAWSSyncAccess"
+	IntegrationRoleTAGPolicy string
+}
+
+// CheckAndSetDefaults ensures the required fields are present.
+func (r *AccessGraphAWSIAMConfigureRequest) CheckAndSetDefaults() error {
+	if r.IntegrationRole == "" {
+		return trace.BadParameter("integration role is required")
+	}
+
+	if r.IntegrationRoleTAGPolicy == "" {
+		r.IntegrationRoleTAGPolicy = defaultPolicyNameForTAGSync
+	}
+
+	return nil
+}
+
+// AccessGraphIAMConfigureClient describes the required methods to create the IAM Policies
+// required for enrolling Access Graph AWS Sync into Teleport.
+type AccessGraphIAMConfigureClient interface {
+	// PutRolePolicy creates or replaces a Policy by its name in a IAM Role.
+	PutRolePolicy(ctx context.Context, params *iam.PutRolePolicyInput, optFns ...func(*iam.Options)) (*iam.PutRolePolicyOutput, error)
+}
+
+type defaultTAGIAMConfigureClient struct {
+	*iam.Client
+}
+
+// NewAccessGraphIAMConfigureClient creates a new TAGIAMConfigureClient.
+func NewAccessGraphIAMConfigureClient(ctx context.Context) (AccessGraphIAMConfigureClient, error) {
+	cfg, err := config.LoadDefaultConfig(ctx, config.WithRegion("" /* region */))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &defaultTAGIAMConfigureClient{
+		Client: iam.NewFromConfig(cfg),
+	}, nil
+}
+
+// ConfigureAccessGraphSyncIAM sets up the roles required for Teleport to be able to pool
+// AWS resources into Teleport.
+// The following actions must be allowed by the IAM Role assigned in the Client.
+//   - iam:PutRolePolicy
+func ConfigureAccessGraphSyncIAM(ctx context.Context, clt AccessGraphIAMConfigureClient, req AccessGraphAWSIAMConfigureRequest) error {
+	if err := req.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	policyDocument, err := awslib.NewPolicyDocument(
+		awslib.StatementAccessGraphAWSSync(),
+	).Marshal()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	_, err = clt.PutRolePolicy(ctx, &iam.PutRolePolicyInput{
+		PolicyName:     &req.IntegrationRoleTAGPolicy,
+		RoleName:       &req.IntegrationRole,
+		PolicyDocument: &policyDocument,
+	})
+	if err != nil {
+		if trace.IsNotFound(awslib.ConvertIAMv2Error(err)) {
+			return trace.NotFound("role %q not found.", req.IntegrationRole)
+		}
+		return trace.Wrap(err)
+	}
+
+	log.Printf("IntegrationRole: IAM Policy %q added to Role %q\n", req.IntegrationRoleTAGPolicy, req.IntegrationRole)
```

Comment:
This is wrong `log` package. (I actually found it in my code and about to create PR for fixing it and couple of other iam config places 😅 )

Reply:
`log` is replaced by `slog` during setup so it's fine to use it anw.